spec/lib/sharepoint/client_spec.rb

```diff
@@ -39,15 +37,27 @@
       end
     end
 
+    context 'with authentication' do
+      [{ value: 'ntlm',  name:   'ntlm' },
+       { value: 'token', name:   'token' }].each do |ocurrence|
+        it "does not raise authentication configuration error for #{ocurrence[:name]} authentication" do
+          correct_config = config
+          correct_config[:authentication] = ocurrence[:value]
+
+          expect do
+            described_class.new(correct_config)
+          end.not_to raise_error(Sharepoint::Errors::InvalidAuthenticationError)
```

Comment:
```suggestion
          end.not_to raise_error
```

Ref:

```
WARNING: Using `expect { }.not_to raise_error(SpecificErrorClass)` risks false positives, since literally any other error would cause the expectation to pass, including those raised by Ruby (e.g. `NoMethodError`, `NameError` and `ArgumentError`), meaning the code you are intending to test may not even get reached. Instead consider using `expect { }.not_to raise_error` or `expect { }.to raise_error(DifferentSpecificErrorClass)`. This message can be suppressed by setting: `RSpec::Expectations.configuration.on_potential_false_positives = :nothing`. Called from /home/runner/work/sharepoint/sharepoint/spec/lib/sharepoint/client_spec.rb:47:in `block (5 levels) in <top (required)>'.
WARNING: Using `expect { }.not_to raise_error(SpecificErrorClass)` risks false positives, since literally any other error would cause the expectation to pass, including those raised by Ruby (e.g. `NoMethodError`, `NameError` and `ArgumentError`), meaning the code you are intending to test may not even get reached. Instead consider using `expect { }.not_to raise_error` or `expect { }.to raise_error(DifferentSpecificErrorClass)`. This message can be suppressed by setting: `RSpec::Expectations.configuration.on_potential_false_positives = :nothing`. Called from /home/runner/work/sharepoint/sharepoint/spec/lib/sharepoint/client_spec.rb:47:in `block (5 levels) in <top (required)>'.
```